src/copilot/app/components/indexing/base.py

```diff
@@ -0,0 +1,271 @@
+"""**Indexing** interface."""
+import logging
+
+from abc import ABC, abstractmethod
+from typing import List, Any
+import aiohttp
+
+from haystack.dataclasses import Document
```

Comment:
Not a good practice to use an implementation specific library in the abstract class. Can we avoid using Document dataclass?